collect_app/src/main/java/org/odk/collect/android/activities/GoogleDriveActivity.java

```diff
@@ -352,6 +287,63 @@ private void getFiles() {
         getFileTask.execute(toDownload);
     }
 
+    @Override
+    protected void updateAdapter() {
+        CharSequence charSequence = getFilterText();
+        filteredList.clear();
+
+        if (charSequence.length() > 0) {
+            for (DriveListItem item : driveList) {
+                if (item.getName().toLowerCase(Locale.US).contains(charSequence.toString().toLowerCase(Locale.US))) {
+                    filteredList.add(item);
+                }
+            }
+        } else {
+            filteredList.addAll(driveList);
+        }
+
+        sortList();
+        if (adapter == null) {
+            adapter = new FileArrayAdapter(this, filteredList);
+            listView.setAdapter(adapter);
+        } else {
+            adapter.notifyDataSetChanged();
+        }
+
+        adapter.notifyDataSetChanged();
+        checkPreviouslyCheckedItems();
+    }
+
+    @Override
+    protected void checkPreviouslyCheckedItems() {
+        listView.clearChoices();
+        for (int i = 0; i < listView.getCount(); i++) {
+            DriveListItem item = (DriveListItem) listView.getAdapter().getItem(i);
+            if (toDownload.contains(item)) {
+                listView.setItemChecked(i, true);
+            }
+        }
+    }
+
+    private void sortList() {
```

Comment:
After sorting the items according to name, you can try adding the sort by type to group the folders and XML forms together.